pandas/core/dtypes/dtypes.py

```diff
@@ -562,21 +580,9 @@ def update_dtype(
         if new_categories is None:
             new_categories = self.categories
 
-        new_ordered = dtype._ordered
-        new_ordered_from_sentinel = dtype._ordered_from_sentinel
+        new_ordered = dtype.ordered
         if new_ordered is None:
-            # maintain existing ordered if new dtype has ordered=None
-            new_ordered = self._ordered
-            if self._ordered and new_ordered_from_sentinel:
-                # only warn if we'd actually change the existing behavior
-                msg = (
-                    "Constructing a CategoricalDtype without specifying "
-                    "`ordered` will default to `ordered=False` in a future "
-                    "version, which will cause the resulting categorical's "
-                    "`ordered` attribute to change to False; `ordered=True` "
-                    "must be explicitly passed in order to be retained"
-                )
-                warnings.warn(msg, FutureWarning, stacklevel=3)
+            new_ordered = self.ordered
```

Comment:
Can you add a comment here why ordered can be None?

Reply:
There's a comment a little further up that actually addresses this (above the `new_categories` definition) but as written it's not immediately noticeable.  I've shortened the logic for both of these to be ternary statements, which hopefully makes the comment is more observable, and cleaned up the wording of the comment.